[Bentimate] iP

Bentimate opened the pull request:
### DukePro

> "All our dreams can come true, if we have the courage to pursue them." ~ Walt Disney
>> "all the best"

DukePro frees your head from unnecessary memory. It's:

- Text-based
- ~~Easy~~ to learn
- Lightweight

**_All you need to do is:_**

1. Download the file from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
2. Run the file
3. add your task according to the instructions
4. Save and quit 🔑 🥇 

I will need to:
- [x] Feed the cats
- [ ] Finish iP
- [ ] Add delight to the experience when all tasks are complete :tada:

some java code: the `main` method.
```java
public static void main(String[] args) {
    Duke duke = new Duke();
    duke.run();
}
```

Commit:
Add Gradle support

Bentimate committed:
Level-1

Bentimate committed:
Level-2

- Added ability to store user input
- Returns stores inputs when prompted
*Also stores null inputs for now

Bentimate committed:
Level-3

-Added mark and unmark methods for tasks
**Have not implemented error handling**

Bentimate committed:
level-3

Bentimate committed:
Level-4
*no error handling yet*

Bentimate committed:
A-TextUITesting

Bentimate committed:
Added error handling for missing information in adding tasks

Bentimate committed:
Added delete function

Bentimate committed:
test commit

jianminglok commented on src/main/java/Data.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.ArrayList;
+
```
Should Javadoc comments be added for this public class?

jianminglok commented on src/main/java/Data.java:
```diff
@@ -0,0 +1,26 @@
+
+public class Data {
+    ArrayList<Task> data = new ArrayList<>();
+
```
Should Javadoc comments be added for this public method?

jianminglok commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+public class Deadline extends Task{
```
Should Javadoc comments be added for this public class?

I noticed the same issue in other classes as well.

jianminglok commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,105 @@
+                char query = input.charAt(input.length() - 1);
+                int pos = Character.getNumericValue(query);
+                //error check for pos exceeding size
+                data.unmark(pos-1);
```
Should an operator be surrounded by a space character?

jianminglok commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,105 @@
+                //error check for pos exceeding size
+                data.unmark(pos-1);
+                System.out.println("Unmarked:");
+                System.out.println(data.getEntry(pos-1).toString());
```
Should an operator be surrounded by a space character?

jianminglok commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,105 @@
+                char query = input.charAt(input.length() - 1);
+                int pos = Character.getNumericValue(query);
+                //error check for pos exceeding size
+                data.mark(pos-1);
```
Should an operator be surrounded by a space character?

This can be found in other parts of the code as well.

jianminglok commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,105 @@
+        return;
+    }
     public static void main(String[] args) {
         String logo = " ____        _        \n"
```
Should the logo String be made a constant?

Bentimate replied:
Good catch,  thanks for that!

jianminglok commented on src/main/java/Data.java:
```diff
@@ -0,0 +1,26 @@
+    public Task getEntry(int pos) {
+        return this.data.get(pos);
+    }
+    public void removeEntry(int pos) { this.data.remove(pos);}
```
Maybe empty lines can be added in between the methods for better readability?

jianminglok reviewed (commented):
I liked how the code is quite clean and well structured overall. I noted a few minor coding standard violations, and perhaps you could consider placing your classes into packages.

Bentimate committed:
no message

zm-l commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,17 @@
+        super.task = dscp;
+    }
+
+    //Override toString
```
Might want to make the comment into a TODO: ... to indicate as a yet to implement method

Bentimate replied:
I see, thanks for the tip!

zm-l commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,17 @@
+public class ToDo extends Task{
+    String tag = "[T]";
+    public ToDo() {
+        super.tag = tag;
```
Should you make this into a `super(tag)` and add a constructor to the Task class?

zm-l commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+public class Event extends Task {
+    String tag = "[E]";
+    protected String from;
```
Is there any consideration to use `protected` rather than `private`?

zm-l commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+public class Event extends Task {
+    String tag = "[E]";
```
Should this be a constant?

zm-l commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,105 @@
+            if (input.contains("delete")) {
+                char query = input.charAt(input.length() - 1);
+                int pos = Character.getNumericValue(query);
+                //error check for pos exceeding size
```
Might be better to move comment related to `pos` on top of it

zm-l commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,105 @@
+                }
+                data.addEntry(todo);
+                System.out.println(String.format("Now you have %d tasks in the list", data.getNum()));
+                continue;
```
Should `continue` be changes to `if else` or `switch` for clarity?

zm-l reviewed (commented):
Overall, your codebase looks good! There are just some minor nits about coding qualities I commented for you to consider, great work so far!

Bentimate committed:
Level-7

Bentimate committed:
Level-8

Bentimate committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/FileManager.java

Bentimate committed:
branch level-7

Bentimate committed:
no message

Bentimate committed:
Merge branch 'branch-Level-7'

Bentimate committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/FileManager.java

Bentimate committed:
Merge branch 'master' of https://github.com/Bentimate/ip

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/FileManager.java

Bentimate committed:
bug fixes to level 7 and 8

Bentimate committed:
A-MoreOOP

Bentimate committed:
A-Packages

Bentimate committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Bentimate committed:
A-Gradle

Bentimate committed:
A-JUnit

Bentimate committed:
Commit for A-JavaDoc

Bentimate committed:
A-CodingStandard

Bentimate committed:
Level-9

Bentimate committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/duke/Parser.java
#	src/main/java/duke/duke/TaskList.java
#	src/main/java/duke/tasks/Task.java

Bentimate committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/duke/Parser.java

Bentimate committed:
bug fixes

Bentimate committed:
A-CheckStyle

Bentimate committed:
Level-10

Bentimate committed:
Add greetings message in GUI

Bentimate committed:
Add greetings message in GUI

Bentimate committed:
Merge branch 'master' of https://github.com/Bentimate/ip

Bentimate committed:
Commit message for Level-9

Add Find command

As the program expands, the user is now able to add more tasks that they want.

In the case where the list of task gets too long, it might be difficult for
the user to find a specific task that they want to view or mark.

Therefore a find command is important as it allows the user
to filter the task list using keywords, allowing them to find their
tasks much faster.

Bentimate committed:
Commit message for Level-10

Add GUI and JavaFX support

As the program grows, text based UI is unable to fully and easily allow
the user to view and edit information.

Therefore a GUI is important as it allows users to view their information,
as well as input data much easily.

It also allows easier debugging as there is now visual indication for any bugs.

Bentimate committed:
A-Assertions

Bentimate committed:
A-CodingQuality

Bentimate committed:
Merge pull request #2 from Bentimate/branch-A-Assertions

A-Assertions

Bentimate committed:
Revert "A-Assertions"

Bentimate committed:
Merge pull request #3 from Bentimate/revert-2-branch-A-Assertions

Revert "A-Assertions"

Bentimate committed:
Merge pull request #4 from Bentimate/branch-A-CodeQuality

A-CodingQuality

Bentimate committed:
Add undo command

Added undo command for adding entries and deleting entries

Bentimate committed:
Add product screenshot

Bentimate committed:
Add a README.md

Bentimate committed:
tried to change branches

Bentimate committed:
Merge branch 'master' of https://github.com/Bentimate/ip

Bentimate committed:
no message
